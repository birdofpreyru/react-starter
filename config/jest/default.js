const config = require('topcoder-react-utils/config/jest/default');

module.exports = config;
