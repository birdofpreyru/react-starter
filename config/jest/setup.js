import '@dr.pogodin/react-utils/config/jest/setup';
