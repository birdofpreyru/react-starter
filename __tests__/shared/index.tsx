/** @jest-environment jsdom */

import Application from 'shared/app';

import { snapshot } from '@dr.pogodin/react-utils/jest';

let mockDevFrontendMode = false;

jest.mock('@dr.pogodin/react-utils', () => {
  const TRU = jest.requireActual('@dr.pogodin/react-utils');
  return {
    ...TRU,
    isomorphy: {
      ...TRU.isomorphy,
      isDevBuild: () => mockDevFrontendMode,
    },
  };
});

// TODO: This does not work now, because new snapshot() implementation
// attempt to render into virtual DOM, which does not quite work unless
// the app is correctly initialized... will do later.
test.skip('Matches shallow snapshot', () => {
  snapshot(<Application />);
});

test.skip('Matches shallow snapshot in dev mode', () => {
  mockDevFrontendMode = true;
  snapshot(<Application />);
});
