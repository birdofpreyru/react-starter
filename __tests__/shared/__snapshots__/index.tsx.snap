// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow snapshot 1`] = `
<MetaTags
  description="Yet another webapp based on Dr. Pogodin's React Starter Kit"
  image={null}
  siteName={null}
  socialDescription={null}
  socialTitle={null}
  title="Dr. Pogodin's React Starter"
  url={null}
>
  <AppRoutes />
</MetaTags>
`;

exports[`Matches shallow snapshot in dev mode 1`] = `
<MetaTags
  description="Yet another webapp based on Dr. Pogodin's React Starter Kit"
  image={null}
  siteName={null}
  socialDescription={null}
  socialTitle={null}
  title="Dr. Pogodin's React Starter"
  url={null}
>
  <AppRoutes />
</MetaTags>
`;
